app/controllers/api/v1/protected/base_resource_controller.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module Protected
+      # This is a JSONAPI-Resources ready controller that tests authentication using JWTSessions and tests
+      # authorization using Pundit policies.
+      class BaseResourceController < ApplicationController
+        include JSONAPI::ActsAsResourceController
+        include Pundit
+
+        # :authorize_access_request! is part of JWTSessions and is all about ensuring valid `authenticated` requests
+        before_action :authorize_access_request!
+
+        # got this far, let's record this request in the `SessionActivity` table
+        before_action :record_session_activity
+
+        private
+
+        # In JSONAPI the `context` function returns a hash that is available at very lifecycle moments in the JSONAPI
```

Comment:
"very lifecycle moments" this is confusing maybe meant something else? "particular lifecycle moments"?